### PR TITLE
[fix](nereids)fix first_value/lead/lag window function bug in nereids

### PR DESCRIPTION
**Pull request by starocean999**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. first_value should be converted to last_value in some case like old planner
2. make the 3rd parameter of lead/lag function support a nullable value in nereids.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

File: be/src/vec/aggregate_functions/aggregate_function_window.h [hunk 1]
```diff
@@ -274,6 +274,8 @@ struct LeadLagData {
                 const auto* nullable_column = assert_cast<const ColumnNullable*>(column);
                 if (nullable_column->is_null_at(0)) {
                     _default_value.reset();
+                } else {
+                    _default_value.set_value(nullable_column->get_nested_column_ptr(), 0);
```
Comment (zhangstar333) [inline be/src/vec/aggregate_functions/aggregate_function_window.h:278]: seems the default value must be const, so if it's nullable_column, the default value must be NULL?

Comment (starocean999) [reply be/src/vec/aggregate_functions/aggregate_function_window.h:278]: new nereids planner may add a cast to literal string internally like:
`select id, create_time, lead(create_time, 1, '2022-09-06 00:00:00') over
                          (order by create_time desc) as "prev_time" from test1; `
->
`select id, create_time, lead(create_time, 1, cast('2022-09-06 00:00:00' as datetime)) over
                          (order by create_time desc) as "prev_time" from test1; `

so, the above code is added for compatibility of nereids planner

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.78 seconds
 stream load tsv:          473 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         38 seconds loaded 2358488459 Bytes, about 59 MB/s
 stream load orc:          69 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230302043840_clickbench_pr_106790.html

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 2]
```diff
@@ -794,12 +794,19 @@ public PlanFragment visitPhysicalWindow(PhysicalWindow<? extends Plan> physicalW
 
         if (partitionExprs.isEmpty() && orderByElements.isEmpty()) {
             if (inputPlanFragment.isPartitioned()) {
-                inputPlanFragment = createParentFragment(inputPlanFragment, DataPartition.UNPARTITIONED, context);
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:797]: when fragment.root is exchange, it is wrong to call createParentFragment(fragment, ...)
How about move this logical into createParentFragment, or remove this function.
it is error prone

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Review by englefly — APPROVED** [review]

**Comment by englefly** [comment]
LGTM

**Pull request merged**